pageserver/src/tenant/timeline.rs

```diff
@@ -82,6 +83,24 @@ enum FlushLoopState {
     Exited,
 }
 
+/// Wrapper for key range to provide reverse ordering by range length for BinaryHeap
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Hole(pub Range<Key>);
+
+impl Ord for Hole {
+    fn cmp(&self, other: &Self) -> Ordering {
+        let other_len = other.0.end.to_i128() - other.0.start.to_i128();
+        let self_len = self.0.end.to_i128() - self.0.start.to_i128();
+        other_len.cmp(&self_len)
```

Comment:
I don't think this should be changed but this is quite subtle that the `other` and `self` are reversed here. For making this a bit more visible there's https://doc.rust-lang.org/stable/std/cmp/struct.Reverse.html and https://doc.rust-lang.org/stable/std/cmp/enum.Ordering.html#method.reverse, latter perhaps more applicable here:

```rust
self_len.cmp(&other_len).reverse()
```

But this is already noted in type documentation.

Reply:
This is still a valid note with the latest key_range => image_coverage changes.

Reply:
Answer in this thread: https://github.com/neondatabase/neon/pull/3597#discussion_r1112150111

Resolving.